### DISCUSSION TITLE
ch7/ch7-12

### DISCUSSION THREAD

**Discussion by giscus** [gopl-zh/gopl-zh.github.com#190] [General]
# ch7/ch7-12

<The Go Programming Language>中文版

https://gopl-zh.github.io/ch7/ch7-12.html

**Comment by Zengyinjun** [comment]
We cannot assume that an arbitrary io.Writer w also has the WriteString method. But we can define a new interface that has just this method and use a type assertion to test whether the dynamic type of w satisfies this new interface.
我们不能对任意io.Writer类型的变量w，假设它也拥有WriteString方法。但是我们可以定义一个只有这个方法的新接口并且使用类型断言来检测是否w的动态类型满足这个新接口。
我们无法假设任意的io.Writer类型的w也具有WriteString方法。

**Reply by Caleb196x** [reply]
真是翻译的让人读起来头大

**Reply by mefly2012** [reply]
就是，机器翻译真烂

**Reply by liveindreamz** [reply]
手里一本机械工业出版社的书，这段翻译的一样烂
